api/core/domain/task_variant.py

```diff
@@ -87,7 +87,8 @@ def validate_output(self, output: Any, partial: bool = False, strip_extras: bool
             )
             return output
         except JSONSchemaValidationError as e:
-            raise JSONSchemaValidationError("Task output does not match schema") from e
+            # Pass through the detailed error message from the enhanced validator
+            raise JSONSchemaValidationError(f"Task output does not match schema: {e}") from e
```

Comment:
that's the real important part of this PR I think and unfortunately there is no test for that portion of the code